build/commands/lib/test.js

```diff
@@ -272,8 +277,11 @@ const runTests = (passthroughArgs, suite, buildConfig, options) => {
         braveArgs,
         runOptions,
       )
-      // Don't run other tests if one has failed already, especially because
-      // this would overwrite the --output file (if given).
+      if (options.output_xml) {
+        // Add filename of xml output of each test suite into the results file
+        fs.appendFileSync(allResultsFilePath, `${testSuite}.xml\n`)
+      }
+      // Don't run other tests if one has failed already.
```

Comment:
this seems wrong and could lead to unnecessary CI cycles because you don't find out about some failures until you've gotten past others.